### PR TITLE
added response hook

### PR DESCRIPTION
**Pull request by ineiti**
for re-doing the simulations of the CoSi-protocol a response-hook is needed. Also removed an unused variable.

### CODE REVIEW

**Comment by nikkolasg** [comment]
PR in https://github.com/dedis/cosi/issues/43

File: protocol/cosi.go [hunk 1]
```diff
@@ -74,6 +75,9 @@ type CommitmentHook func(in []abstract.Point) error
 // challenge is received
 type ChallengeHook func(ch abstract.Scalar) error
 
+// ResponseHook allows registering a handler when the response is calculated
```
Comment (nikkolasg) [inline protocol/cosi.go]: The response hook is not receiving the calculated response but the children's response.

Comment (ineiti) [inline protocol/cosi.go]: same as the commit-hook. I think they should behave the same, shouldn't they?

Comment (nikkolasg) [inline protocol/cosi.go]: Yep the commitment hook receives the children's commitment, so it's the same. But the comment say here `when the response is calculated`  =>  the response is calculated after.

Comment (ineiti) [inline protocol/cosi.go]: will change the comment then, as I only need the children's response - the total response is given with `SignatureHandler`.

**Comment by nikkolasg** [comment]
OK to merge

**Pull request merged**